### ISSUE TITLE
Open folder + file explorer

### ISSUE THREAD

**Issue by RednibCoding** [bmx-ng/maxide#34]
It would be nice if you had the option to open an entire folder (File->Open Folder) and a file explorer on the left side, so you have your project root folder as root node like in Sublimetext for example. So you would be able to open/rename/create/delete bmx files from this view.
One could add the option to enable/disable this gadget, so everybody could decide if they want to see it or not.

![Openfolder](https://user-images.githubusercontent.com/28296119/54077844-665b7500-42be-11e9-9d65-e3de2f6af2b7.png)
![Fileexplorer](https://user-images.githubusercontent.com/28296119/54077848-72473700-42be-11e9-839d-c43eef85f20c.png)

**Comment by GWRon** [comment]
I most of the time have .bmx files of multiple folders open in my editors. Should the "project folder pane" then switch to the folder of the currently selected file? What happens if you went into a subdirectory, switched to another file - and switch back. It would go to the files directory again - dismissing your subdirectory traversal.

If you do not want to use the file's project folder - which then to use for multiple opened project files (from different folders) ?


Easiest of course is to have some kind of "use current folder on startup" + a button/context menu/... "go to current files folder".


general: You can have a "file explorer"-like thing by utilizing "TreeFolder" or "TreeView"-kind widgets. Still you have to write all the interaction stuff (move, create, copy, ...) and you also have to handle image icons, app icons ... 
There is no ready-made "file explorer" widget for MaxGUI - which has to be cross-platform too.

Things might differ with an qt or wx-based IDE.

**Comment by RednibCoding** [comment]
No, there is no need for fancy icons or a switching back an forth. It should just be in like sublime, where you select a folder and it shows you all the files and subdirectories. From there you can double-click a file to open it. But the root node of this folder-view will always be the one folder you have chosen to open at the beginning. Its a static treeview with the one root folder/node you have chosen. From there you can expand folder nodes and double-click as many files you like to open, the treeview stays the same.

Like in the picture shown below, i have opened several files from different directories but the root/folder (that i have chosen to open via File->Open Folder) stays the same.

![StaticTreeview](https://user-images.githubusercontent.com/28296119/54078208-9d348980-42c4-11e9-9ed1-598595b37840.png)

**Comment by GWRon** [comment]
Yes and what happens if you add a folder/file in that directory from within another file explorer ?

A Treeview widget has no inbuilt logic. MaxIDE would need to observe folder changes on its own and update its folder treeview then.
Handling renaming needs to get coded too and so on.


Time is better invested in writing a language plugin for a more advanced IDE and keep MaxIDE simple.
